Description: Remove Bash assumptions
 The example baraction.sh script shipped in spectrwm's release relies on
 some non-POSIX shell features.
 .
 This patch modifies the script so that it is possible to run it on a
 POSIX-compliant shell. The patched script still relies on a bunch of
 OpenBSD-specific features, which are taken care of in a different patch.
Bug-Debian: http://bugs.debian.org/530990
Forwarded: Marco Peereboom <marco@peereboom.us>
Author: Andrea Bolognani <eof@kiyuko.org>
Last-Update: 2010-07-21
Index: spectrwm.git/baraction.sh
===================================================================
--- spectrwm.git.orig/baraction.sh	2012-02-19 19:34:10.263426784 +0100
+++ spectrwm.git/baraction.sh	2012-02-19 19:44:08.263671744 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 #
 
+APM="/usr/sbin/apm -alb"
+IOSTAT="/usr/sbin/iostat -c -C 3600"   # wish infinity was an option
+TOP="/usr/bin/top"
+
 print_date() {
 	# The date is printed to the status bar by default.
 	# To print the date through this script, set clock_enabled to 0
@@ -11,7 +15,7 @@
 }
 
 print_mem() {
-	MEM=`/usr/bin/top | grep Free: | cut -d " " -f6`
+	MEM=`${TOP} | grep Free: | cut -d " " -f6`
 	echo -n "Free mem: $MEM  "
 }
 
@@ -83,19 +87,20 @@
 while :; do
 	# instead of sleeping, use iostat as the update timer.
 	# cache the output of apm(8), no need to call that every second.
-	/usr/sbin/iostat -C -c 3600 |&	# wish infinity was an option
-	PID="$!"
-	APM_DATA=""
 	I=0
-	trap "kill $PID; exit" TERM
-	while read -p; do
-		if [ $(( ${I} % 1 )) -eq 0 ]; then
-			APM_DATA=`/usr/sbin/apm -alb`
+	${IOSTAT} 2>&1 |
+	while read IOSTAT_DATA; do
+		if [ ${I} -eq 0 ]; then
+			PID="$!"
+			trap "kill $PID; exit" TERM
+		fi
+		if [ $(( ${I} % 10 )) -eq 0 ]; then
+			APM_DATA=`${APM}`
 		fi
 		if [ $I -gt 2 ]; then
 			# print_date
-			print_mem $MEM
-			print_cpu $REPLY
+			print_mem
+			print_cpu $IOSTAT_DATA
 			print_cpuspeed
 			print_apm $APM_DATA
 			echo ""
